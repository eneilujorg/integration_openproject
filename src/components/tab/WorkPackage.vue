<template>
	<div class="workpackage">
		<div class="row">
			<div class="row__status"
				:style="{'background-color':workpackage.statusCol}">
				<div class="row__status__title">
					{{ workpackage.statusTitle }}
				</div>
			</div>
			<div class="row__workpackage">
				#{{ workpackage.id }} - {{ workpackage.project }}
			</div>
		</div>
		<div class="row">
			<div class="row__subject">
				<span class="row__subject__type" :style="{'color':workpackage.typeCol}">
					{{ workpackage.typeTitle }}
				</span>
				{{ workpackage.subject }}
			</div>
		</div>
		<div class="row">
			<div v-if="workpackage.assignee" class="row__assignee">
				<div class="row__assignee__avatar">
					<Avatar class="item-avatar"
						:size="23"
						:url="workpackage.picture"
						:user="workpackage.assignee"
						:display-name="workpackage.assignee" />
				</div>
				<div class="row__assignee__assignee">
					{{ workpackage.assignee }}
				</div>
			</div>
		</div>
	</div>
</template>

<script>
import Avatar from '@nextcloud/vue/dist/Components/Avatar'

export default {
	name: 'WorkPackage',
	components: {
		Avatar,
	},
	props: {
		workpackage: {
			type: Object,
			required: true,
		},
	},
}
</script>

<style scoped lang="scss">
.workpackage {
	width: 100%;
	padding: 15px 6px 0 10px;
	border-bottom: 1px solid rgb(237 237 237);

	.row {
		display: flex;
		padding: 3px;
		flex-wrap: wrap;

		&__status {
			padding: 5px 10px;
			width: fit-content;
			height: 20px;
			justify-content: center;
			display: flex;
			align-items: center;
			text-align: center;
			font-size: 0.75rem;
			border-radius: 2px;
			margin-right: 4px;

			&__title {
				font-size: 0.75rem;
				line-height: 14px;
				text-align: center;
				filter: contrast(0) brightness(0);
			}
		}

		&__workpackage {
			color: #878787;
			font-size: 0.8rem;
			height: 20px;
			line-height: 20px;
		}

		&__subject {
			font-size: 0.875rem;
			text-align: justify;
			white-space: normal;
			text-justify: inter-word;
			display: -webkit-box;
			-webkit-line-clamp: 2;
			-webkit-box-orient: vertical;
			overflow: hidden;
			text-overflow: ellipsis;

			&__type {
				font-size: 0.75rem;
				font-weight: bold;
				text-transform: uppercase;
				margin-right: 4px;
			}
		}

		&__assignee {
			display: flex;
			flex-direction: row;
			padding: 5px;

			&__assignee {
				font-size: 0.81rem;
				color: #878787;
				text-align: center;
				padding-left: 5px;
			}
		}
	}
}

.workpackage div {
	cursor: pointer;
}
</style>
