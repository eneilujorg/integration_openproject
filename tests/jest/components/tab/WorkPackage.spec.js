/**
 * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2022 Jankari Tech Pvt. Ltd.
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import { createLocalVue, shallowMount } from '@vue/test-utils'

import WorkPackage from '../../../../src/components/tab/WorkPackage.vue'
import workPackagesSearchResponse from '../../fixtures/workPackagesSearchResponse.json'

const localVue = createLocalVue()

describe('WorkPackage.vue', () => {
	let wrapper
	const workPackagesSelector = '.workpackage'
	beforeEach(() => {

		wrapper = shallowMount(WorkPackage, {
			localVue,
			propsData: {
				workpackage: workPackagesSearchResponse[0],
			},
		})
	})
	it('shows work packages information', async () => {
		const workPackages = wrapper.find(workPackagesSelector)
		expect(workPackages.exists()).toBeTruthy()
		expect(workPackages).toMatchSnapshot()

	})
})
