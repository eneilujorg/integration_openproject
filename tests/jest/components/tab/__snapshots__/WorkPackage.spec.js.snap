// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsTab.vue Test shows work packages information 1`] = `
<div class="workpackage">
  <div class="row">
    <div class="row__status" style="background-color: blue;">
      <div class="row__status__title">
        in-progress
      </div>
    </div>
    <div class="row__workpackage">
      #1 - test
    </div>
  </div>
  <div class="row">
    <div class="row__subject"><span class="row__subject__type" style="color: red;">
				task
			</span>
      Organize work-packages
    </div>
  </div>
  <div class="row">
    <div class="row__assignee">
      <div class="row__assignee__avatar">
        <avatar-stub url="/server/index.php/apps/integration_openproject/avatar?userId=1&amp;userName=System" user="test" showuserstatus="true" showuserstatuscompact="true" displayname="test" size="23" allowplaceholder="true" menuposition="center" menucontainer="body" class="item-avatar"></avatar-stub>
      </div>
      <div class="row__assignee__assignee">
        test
      </div>
    </div>
  </div>
</div>
`;
